core/synchronization/src/lib.rs

```diff
@@ -118,171 +90,105 @@ impl<T: SyncAdapter> Synchronization<T> {
-
         log::info!("[sync]finish");
-
         Ok(())
     }
-
     async fn try_create_consume_info_table(&self) -> Result<()> {
         let mut conn = self.pool.acquire().await?;
         let _ = sql::create_consume_info_table(&mut conn).await;
         Ok(())
     }
 
-    async fn sync_batch_insert(&self, chain_tip: u64, sync_list: Vec<u64>) {
-        log::info!(
-            "[sync] chain tip is {}, need sync {}",
-            chain_tip,
-            sync_list.len()
-        );
-
-        for set in sync_list.chunks(self.sync_task_size) {
-            let sync_set = set.to_vec();
-            let (rdb, adapter) = (self.pool.clone(), Arc::clone(&self.adapter));
-
-            loop {
-                let task_num = current_task_count();
-                if task_num < self.max_task_number {
-                    add_one_task();
-                    tokio::spawn(async move {
-                        sync_process(sync_set, rdb, adapter).await;
-                    });
-
-                    break;
-                } else {
-                    sleep(Duration::from_secs(5)).await;
-                }
-            }
-        }
-    }
-
-    async fn build_to_sync_list(&self) -> Result<Vec<BlockNumber>> {
-        let mut to_sync_number_set = (0..=self.chain_tip).collect::<HashSet<_>>();
-        let sync_completed_set = self.get_sync_completed_numbers().await?;
-        sync_completed_set.iter().for_each(|num| {
-            to_sync_number_set.remove(num);
-        });
-
-        Ok(to_sync_number_set.into_iter().collect())
-    }
-
-    async fn get_sync_completed_numbers(&self) -> Result<Vec<BlockNumber>> {
-        let mut conn = self.pool.acquire().await?;
-        let res = sql::get_sync_completed_numbers(&mut conn).await?;
-        Ok(res.iter().map(|t| t.block_number).collect())
-    }
+    async fn sync_metadata(&self) {
+        log::info!("[sync] chain tip is {}", self.chain_tip);
 
-    async fn build_to_sync_indexer_list(&self) -> Result<Vec<BlockNumber>> {
-        let db_tip = RelationalStorage {
-            pool: self.pool.clone(),
-        }
-        .get_tip_number()
-        .await;
-
-        let db_tip = if let Ok(db_tip) = db_tip {
-            db_tip
-        } else {
-            return Ok(vec![]);
-        };
-
-        let mut to_sync_number_set = (0..=db_tip).collect::<HashSet<_>>();
-        let sync_completed_set = self.get_sync_indexer_completed_numbers().await?;
-        sync_completed_set.iter().for_each(|num| {
-            to_sync_number_set.remove(num);
-        });
-
-        log::info!("[sync] to sync indexer cell {}", to_sync_number_set.len());
-        Ok(to_sync_number_set.into_iter().collect())
-    }
-
-    async fn get_sync_indexer_completed_numbers(&self) -> Result<Vec<BlockNumber>> {
-        let mut conn = self.pool.acquire().await?;
-        let res = conn.fetch_list::<SyncStatus>().await?;
+        for id in (0..self.chain_tip).step_by(TASK_LEN as usize) {
```

Comment:
should be `=self.chain_tip` ?
```suggestion
        for id in (0..=self.chain_tip).step_by(TASK_LEN as usize) {
```